docs/_articles/en/v3.x/code-analyzer-vs-code-extension.md

```diff
@@ -0,0 +1,144 @@
+--- title: Salesforce Code Analyzer Visual Studio Code Extension (Beta) lang: en redirect_from: /en/code-analyzer-vs-code-extension ---
+
+The Salesforce Code Analyzer (Code Analyzer) Visual Studio (VS) Code extension is an extension that integrates many of Code Analyzer’s most useful features into VS Code, allowing them to be easily run with clicks instead of terminal commands.
+
+> **_NOTE:_** If you’re listing a managed package on AppExchange, it must pass security review. You’re also required to upload your Salesforce Code Analyzer scan reports. Run Code Analyzer via the VS Code extension and update your code. Next, to produce the required scan reports for your AppExchange listing, you must run Code Analyzer via the command line either within VS Code or as standalone. Attach your scan reports to your submission in the AppExchange Security Review Wizard. Read [Scan Your Solution with Salesforce Code Analyzer](https://developer.salesforce.com/docs/atlas.en-us.packagingGuide.meta/packagingGuide/security_review_code_analyzer_scan.htm) for details.
+
+## Using Code Analyzer VS Code Extension (beta)
+
+Use Code Analyzer VS Code extension (beta) to scan multiple languages:
+
+* [PMD rule engine](https://pmd.github.io/)
+* [RetireJS](https://retirejs.github.io/retire.js/)
+* [Salesforce Graph Engine](https://forcedotcom.github.io/sfdx-scanner/en/v3.x/salesforce-graph-engine/introduction/) (Generally Available rules only)
+
+Additionhally, you can also [enable]([https://forcedotcom.github.io/sfdx-scanner/en/v3.x/faq/#q-how-do-i-enable-engine-xs-default-rules-for-language-y]) these languages in Salesforce Code Analyzer settings:
```

Comment:
@jfeingold35 I updated this sentence per your comment. But I'm confused: if the user wants to run Code Analyzer on Java or XML, can they use the VS Code extension? If they have to change the setting via the CLI and can't use the VS Code extension, then this paragraph and two bullet points should be removed. Also, see the thread on the core release note - it's near ready to check in, so I'd appreciate your input there, too. 

https://docs.google.com/document/d/1DrCR4CAgM0yrlbmCi3eWnbtoZlDSupXc2n4pvbLfXoY/edit?usp=sharing